httpx/_models.py

```diff
@@ -332,7 +332,13 @@ def __init__(
             Cookies(cookies).set_cookie_header(self)
 
         if stream is None:
-            headers, stream = encode_request(content, data, files, json)
+            headers, stream = encode_request(
+                content=content,
+                data=data,
+                files=files,
+                json=json,
+                content_type=self.headers.get("content-type"),
+            )
```

Comment:
By the comment on the other thread, I mean, why not write:
```suggestion
            headers, stream = encode_request(
                content=content,
                data=data,
                files=files,
                json=json,
                boundary= get_multipart_boundary_from_content_type(self.headers.get("content-type")),
            )
```

That way, we do not have function signatures that have both a `boundary` and a `content_type` that could possibly have contradictory data?

Reply:
By which I mean, the `content_type` argument could be removed from `encode_request` and `encode_multipart_data`?

Reply:
Okay I liked it once I wrote it, I think given that `encode_request` already takes a `boundary` parameter (which only makes sense in the context of multipart requests) moving the call to where you are suggesting is not a big deal

Reply:
Yes, that kind of was my thought (if `encode_request` did not have `boundary` I would not have suggested that). Thank you for accepting my suggestion.

I just wonder if you would also go also look at potentially reviewing the remaining changes in `httpx/_content.py`? I mean, these changes seem mostly vestigial to me (e.g. the `content_type` in `encode_request`, and the changes in `encode_multipart_data` are now purely of variable and formatting (and `boundary` is now a required argument)

Reply:
done